lib/galaxy/tools/search/__init__.py

```diff
@@ -170,17 +172,23 @@ def search(self, q, tool_name_boost, tool_section_boost,
         # would have a higher score than a document with 'bowtie arthur' which is usually unexpected for a user.
         # Hence we introduce a bonus on multi-hits using the 'factory()' method using a scaling factor between 0-1.
         # https://whoosh.readthedocs.io/en/latest/parsing.html#searching-for-any-terms-instead-of-all-terms-by-default
+        # Adding the FuzzyTermPlugin to account for misspellings and typos, using a max distance of 2
         og = OrGroup.factory(0.9)
-        self.parser = MultifieldParser(['name', 'description', 'section', 'help', 'labels', 'stub'], schema=self.schema, group=og)
+        self.parser = MultifieldParser(['name', 'old_id', 'description', 'section', 'help', 'labels', 'stub'], schema=self.schema, group=og)  # , termclass=FuzzyTerm)
+        self.parser.add_plugin(FuzzyTermPlugin())
+
         cleaned_query = q.lower()
         if tool_enable_ngram_search is True:
             rval = self._search_ngrams(cleaned_query, tool_ngram_minsize, tool_ngram_maxsize, tool_search_limit)
             return rval
         else:
             cleaned_query = ' '.join(token.text for token in self.rex(cleaned_query))
             # Use asterisk Whoosh wildcard so e.g. 'bow' easily matches 'bowtie'
-            parsed_query = self.parser.parse(cleaned_query + '*')
+            #parsed_query = self.parser.parse('*' + cleaned_query + '*' + '~2')
+            parsed_query = self.parser.parse('*' + cleaned_query + '*' + ' ' + cleaned_query + '~2')
```

Comment:
Why are there both queries in there ? Seems like you'd only want the commented one above ?
I'd really prefer not to switch `FuzzyTermPlugin`. We've switched the default to `tool_enable_ngram_search`, which handles fuzzy searches as well.
If that's ok with you I'd suggest keeping that as a follow-up if we discover problems with the ngram search ?